linux_os/guide/services/ssh/ssh_server/firewalld_sshd_port_enabled/oval/shared.xml

```diff
@@ -1,109 +1,209 @@
+    </unix:file_object>
+
+    <!-- During the refactoring of this file, it was tried many techniques to use the available
+         OVAL resources in order to detect and assess only active zone, which are zones with at
+         least one NIC assigned to it. Since it was possible to easily have the list of active
+         zones, it was cumbersome to use that list in other OVAL objects without introduce a high
+         level of complexity to make sure environments with multiple NICs and multiple zones are
+         in use. So, in favor of simplicity and readbility it was decided to work with a static
+         list. It means that, in the future, it is possible this list needs to be updated. -->
+    <local_variable id="var_firewalld_sshd_port_enabled_default_zones" version="1"
+        datatype="string"
+        comment="Regex containing the list of zones files delivered in the firewalld package">
+        <literal_component>^(dmz|external|home|internal|public|trusted|work)\.xml$</literal_component>
+    </local_variable>
+
+    <!-- If any default zone is modified by the administrator, the respective zone file is placed
+         in the /etc/firewalld/zones dir in order to override the default zone settings. The same
+         directory is applicable for new zones created by the administrator. Therefore, all files
+         in this directory should also allow SSH. -->
+    <ind:xmlfilecontent_test id="test_firewalld_sshd_port_enabled_zone_ssh_enabled_etc"
+        check="all" check_existence="at_least_one_exists" version="1"
+        comment="SSH service is defined in all zones created or modified by the administrator">
+      <ind:object object_ref="object_firewalld_sshd_port_enabled_zone_files_etc"/>
+      <ind:state state_ref="state_firewalld_sshd_port_enabled_zone_files_etc"/>
+    </ind:xmlfilecontent_test>
+
+    <ind:xmlfilecontent_object id="object_firewalld_sshd_port_enabled_zone_files_etc" version="1">
+      <ind:path>/etc/firewalld/zones</ind:path>
+      <ind:filename operation="pattern match">^.*\.xml$</ind:filename>
+      <ind:xpath>/zone/service[@name='ssh']</ind:xpath>
+    </ind:xmlfilecontent_object>
+
+    <ind:xmlfilecontent_state id="state_firewalld_sshd_port_enabled_zone_files_etc" version="1">
+      <ind:xpath>/zone/service[@name='ssh']</ind:xpath>
+    </ind:xmlfilecontent_state>
+
+    <!-- SSH service is configured as expected -->
+    <!-- The firewalld package brings many services already defined out-of-box, inclusing SSH.
+         SSH port is defined as 22/tcp by default. However, is that possible cases where the
+         administrator want, by any reason, to change this default port. The proper way to do this
+         is overriding the respectice SSH service file by populating the /etc/firewalld/services
+         directory. So, its necessary to ensure the default file is interger and the, if is the
+         case, the customized service is properly configured. -->
```

Comment:
What does it mean `the default file is interger`? I thought there is a type and the word should be `integer` but it still doesn't make sense to me.

Also there is a unnecessary  `the` in the sentence (the one before comma).

Reply:
Indeed there is a typo there. Thanks for noticing.
By `default file` it means the file in `/usr/lib/firewalld` dir, which is part of the package. This file should not be modified. If there is a intention to modify the default ssh port, the proper way is to create a modified file with the same name within `/etc/firewalld`.

By the way, I believe this comment was created very early, when I started to work on this rule and I didn't review it before sending the PR. I think its wording can be improved too.

Reply:
If the wording could be improved, it would be great :)
Not saying the wording is bad, but it would make it easier for person who doesn't have knowledge of the topic.